sycl/doc/extensions/proposed/sycl_ext_intel_fpga_annotated_arg_properties.asciidoc

```diff
@@ -0,0 +1,280 @@
+
+template <typename T, typename PropertyListT>
+struct is_property_key_of<register_map_key,
+  annotated_arg<T, PropertyListT>> : std::true_type {};
+
+struct conduit_key {
+  using value_t = property_value<conduit_key>;
+};
+
+inline constexpr conduit_key::value_t conduit;
+
+template<> struct is_property_key<conduit_key> : std::true_type {};
+
+template <typename T, typename PropertyListT>
+struct is_property_key_of<conduit_key,
+  annotated_arg<T, PropertyListT>> : std::true_type {};
+
+struct stable_key {
+  using value_t = property_value<stable_key>;
+};
+
+inline constexpr stable_key::value_t stable;
+
+template<> struct is_property_key<stable_key> : std::true_type {};
+
+template <typename T, typename PropertyListT>
+struct is_property_key_of<stable_key,
+  annotated_arg<T, PropertyListT>> : std::true_type {};
+} // namespace experimental::oneapi::ext::sycl
+```
+--
+
+[frame="topbot",options="header"]
+|===
+|Property |Description
+
+a|
+[source,c++]
+----
+conduit
+----
+a|
+Directs the compiler to create a dedicated input port on the kernel for the
+input data.
+
+a|
+[source,c++]
+----
+register_map
+----
+a|
+Directs the compiler to create a register to store the base address of the
+of the pointer interface as opposed to creating a dedicated input port on the
+kernel for supplying the pointer base address.
+
+a|
+[source,c++]
+----
+stable
+----
+a|
+Specifies that the input pointer address to the kernel will not change during
+the execution of the kernel. The input can still change after all active
+kernel invocations have finished.
+
+If the input is changed while the kernel is executing, the behavior is
+undefined.
+
+|===
+--
+
+=== Aliases provided for convenience
+
+[source,c++]
+----
+namespace sycl::ext::oneapi::experimental{
+  template <typename T, typename PropertyListT>
+  using register_map = annotated_arg<T, properties{
+    register_map, PropertyListT}>;
+
+  template <typename T, typename PropertyListT>
+  using conduit = annotated_arg<T, properties{
+    conduit, PropertyListT}>;
+}; // namespace sycl::ext::oneapi::experimental
+----
+
+=== Usage Examples
+
+The examples below show a simple kernel with two integer arguments marked with
+`register_map` and `stable` properties.
+
+.Usage example with a SYCL functor
+```c++
+using sycl::ext::oneapi::experimental;
+struct MyKernel {
+  using RegisterMapArg = annotated_arg<int, properties<register_map, stable>>;
+  RegisterMapArg a;
+  RegisterMapArg b;
+  ...
+  void operator()() const {
+    ... = a * b;
+  }
+};
+
+int main () {
+  sycl::queue q;
+  int data_a = ...
+  int data_b = ...
+  
+  MyKernel my_k;
+  my_k.a = data_a;
+  my_k.a = data_b;
+  ...
+  q.single_task(my_k).wait();
+  ...
+}
+```
+
+.Usage example with a SYCL lambda
+```c++
+using sycl::ext::oneapi::experimental;
+
+int main () {
+  sycl::queue q;
+  int data_a = ...
+  int data_b = ...
+  auto a = annotated_arg(data_a, properties{register_map, stable});
```

Comment:
It could be interesting to add an `operator()` to properties to use them as decorator to be able just to write:
```suggestion
  auto a = properties{register_map, stable}(data_a);
```
Some other intereting ideas in [SIMD in C++20 - EVE of a new Era - Joël Falcou - CPPP 2021](https://www.youtube.com/watch?v=Tqnz4IsmIFY)